core/controllers/question_editor.py

```diff
@@ -25,12 +25,47 @@
 import feconf
 
 
+class QuestionCreationHandler(base.BaseHandler):
+    """A handler that creates the question model given a question dict."""
+
+    @acl_decorators.can_manage_question_skill_status
+    def get(self):
+        """Handles GET requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        new_question_id = question_services.get_new_question_id()
+        self.render_json({
+            'questionId': new_question_id
+        })
+
+    @acl_decorators.can_manage_question_skill_status
+    def post(self):
+        """Handles POST requests."""
+        if not feconf.ENABLE_NEW_STRUCTURES:
+            raise self.PageNotFoundException
+
+        question_dict = self.payload.get('question_dict')
+        if (
+                ('id' not in question_dict) or
```

Comment:
The incoming dict doesn't need an ID. Otherwise you run into the possible race condition where:

- person A gets ID 1 and creates a question
- person B (by fluke) also gets ID 1 and creates a question
- person A submits the question
- person B submits the question, overwriting person A's question ... oops!

So, only assign the ID at this point; you can have it be None beforehand. Also, in general you might want to have a check that the question ID you're adding is not already in the datastore, before adding it.

Reply:
Done.